workers/generic-worker/build.sh

```diff
@@ -60,13 +60,25 @@ fi
 
 function install {
   if ! $PUBLISH; then
-      GOOS="${2}" GOARCH="${3}" CGO_ENABLED=0 go install -ldflags "-X main.revision=$(git rev-parse HEAD)" -tags "${1}" -v ./...
+      GOOS="${2}" GOARCH="${3}" CGO_ENABLED=0 go install -tags "${1}" -v ./...
       GOOS="${2}" GOARCH="${3}" go vet -tags "${1}" ./...
       # note, this just builds tests, it doesn't run them!
       GOOS="${2}" GOARCH="${3}" CGO_ENABLED=0 go test -tags "${1}" -c .
       GOOS="${2}" GOARCH="${3}" CGO_ENABLED=0 go test -tags "${1}" -c ./livelog
   fi
-  GOOS="${2}" GOARCH="${3}" CGO_ENABLED=0 go build -o "$OUTPUT_DIR/generic-worker-${1}-${2}-${3}" -ldflags "-X main.revision=$(git rev-parse HEAD)" -tags "${1}" -v .
+  GOOS="${2}" GOARCH="${3}" CGO_ENABLED=0 go build -o "$OUTPUT_DIR/generic-worker-${1}-${2}-${3}" -ldflags "-X github.com/taskcluster/taskcluster/v30/workers/generic-worker.revision=$(git rev-parse HEAD)" -tags "${1}" -v .
+
+  # verify that the revision has been correctly embedded when publishing
+  if $PUBLISH; then
+      if [ "${2}" = "linux" -a "${3}" = "amd64" ]; then
+          vers=$("$OUTPUT_DIR/generic-worker-${1}-${2}-${3}" --version \
+              | grep -E 'revision: https://github.com/taskcluster/taskcluster/commits/[a-z0-9]{40}')
+          if [ -z "$vers" ]; then
+              echo "The --version option does not output a proper revision link"
+              exit 1
+          fi
+      fi
+  fi
```

Comment:
Since a developer can break this at any time, we should find out immediately, not only when a release is made, so let's lose this outer `if` statement. Note, if you had done this, you would have noticed that you had broken it yourself. 😉 

This version is a little bit simpler, and would catch the issue as soon as it breaks:

```suggestion
  if [ "${2}" == "$(go env GOOS)" ] && [ "${3}" == "$(go env GOARCH)" ]; then
    if ! "$OUTPUT_DIR/generic-worker-${1}-${2}-${3}" --version | \
    grep -Eq 'revision: https://github.com/taskcluster/taskcluster/commits/[a-z0-9]{40}'; then
      echo "The --version option does not output a proper revision link"
      exit 1
    fi
  fi
```